src/joystick/hidapi/SDL_hidapi_gip.c

```diff
@@ -0,0 +1,2401 @@
+    attachment &= GIP_FLAG_ATTACHMENT_MASK;
+    return GIP_SendSystemMessage(device, GIP_CMD_SET_DEVICE_STATE, attachment, buffer, sizeof(buffer));
+}
+
+static bool GIP_SendInitSequence(GIP_Device *device)
+{
+    if (device->device->vendor_id == USB_VENDOR_MICROSOFT &&
+        device->device->product_id == USB_PRODUCT_XBOX_ONE_ELITE_SERIES_2)
+    {
+        /*
+         * The meaning of this packet is unknown and not documented, but it's
+         * needed for the Elite 2 controller to start up on older firmwares
+         */
+        static const Uint8 set_device_state[] = { GIP_STATE_UNK6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x55, 0x53, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0 };
+
+        if (!GIP_SendSystemMessage(device,
+            GIP_CMD_SET_DEVICE_STATE,
+            0,
+            set_device_state,
+            sizeof(set_device_state)))
+        {
+            return false;
+        }
+
+        if (!GIP_EnableEliteButtons(device)) {
+            return false;
+        }
+    }
+    if (!GIP_SendSetDeviceState(device, GIP_STATE_START, 0)) {
+        return false;
+    }
+    device->device_state = GIP_STATE_START;
+
+    if (!GIP_SendGuideButtonLED(device, GIP_LED_GUIDE_ON, 20)) {
+        return false;
+    }
+
+    if (GIP_SupportsSystemMessage(device, GIP_CMD_SECURITY, false) &&
+        !(device->features & GIP_FEATURE_SECURITY_OPT_OUT))
+    {
+        /* TODO: Implement Security command property */
+        Uint8 buffer[] = { 0x1, 0x0 };
+        GIP_SendSystemMessage(device, GIP_CMD_SECURITY, 0, buffer, sizeof(buffer));
+    }
+
+    if (GIP_SupportsVendorMessage(device, GIP_CMD_INITIAL_REPORTS_REQUEST, false)) {
+        GIP_InitialReportsRequest request = { 0 };
+        GIP_SendVendorMessage(device, GIP_CMD_INITIAL_REPORTS_REQUEST, 0, (const Uint8 *)&request, sizeof(request));
+    }
+    return HIDAPI_JoystickConnected(device->device, NULL);
+}
+
+static bool GIP_EnsureMetadata(GIP_Device *device)
+{
+        
+    switch (device->got_metadata) {
+    case GIP_METADATA_GOT:
+    case GIP_METADATA_FAKED:
+        return true;
+    case GIP_METADATA_NONE:
+        if (device->quirks & GIP_QUIRK_BROKEN_METADATA) {
+            GIP_SendSystemMessage(device, GIP_CMD_METADATA, 0, NULL, 0);
+            GIP_SetMetadataDefaults(device);
+            return GIP_SendInitSequence(device);
+        } else if (device->got_hello) {
+            device->got_metadata = GIP_METADATA_PENDING;
+            device->metadata_next = SDL_GetTicks() + 500;
+            device->metadata_retries = 0;
+            return GIP_SendSystemMessage(device, GIP_CMD_METADATA, 0, NULL, 0);
+        } else {
+            return GIP_SetMetadataDefaults(device);
+        }
+    default:
+        return true;
+    }
+}
+
+static bool GIP_HandleCommandProtocolControl(
+    GIP_Device *device,
+    const GIP_Header *header,
+    const Uint8 *bytes,
+    int num_bytes)
+{
+    // TODO
+    SDL_LogDebug(SDL_LOG_CATEGORY_INPUT, "GIP: Unimplemented Protocol Control message");
```

Comment:
This seems fine, alternatively you could bump it up to LogWarn() and use `#ifdef DEBUG_XBOX_PROTOCOL`

Reply:
I don't think it's a big enough deal to add more ifdef gating.